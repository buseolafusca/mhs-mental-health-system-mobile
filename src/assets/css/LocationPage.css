.SurveyResult1{
    padding-bottom: 0.5rem;  
    min-height: calc(100vh - 200px);
    position: relative;
  }