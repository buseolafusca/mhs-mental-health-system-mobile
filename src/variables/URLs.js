var nhsUrl = 'http://mhsbackend.azurewebsites.net'
var baseUrl = 'http://178.128.34.125'
var fetchQuestionnaireUrl = '/api/v1/questionnaire_sJS/'
var answersUrl = '/api/v1/patientanswers/'
var registerUrl = '/patients/register'
var serviceUrl = '/api/v1/organisations/services/'

// HERE Related API
var getLocationByPostCodeUrl = 'https://geocoder.api.here.com/6.2/geocode.json'
var getCategoriesByLocationUrl = 'https://places.cit.api.here.com/places/v1/categories/places'
var getPlacesByCategoryLocationUrl = 'https://places.cit.api.here.com/places/v1/discover/explore'

module.exports = {
  nhsUrl,
  baseUrl,
  fetchQuestionnaireUrl,
  answersUrl,
  getLocationByPostCodeUrl,
  getCategoriesByLocationUrl,
  getPlacesByCategoryLocationUrl,
  registerUrl,
  serviceUrl
}
