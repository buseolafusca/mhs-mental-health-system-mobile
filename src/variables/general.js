
var baseUrl = "http://mhsbackend.azurewebsites.net";
var fetchQuestionnaireUrl = "/api/v1/questionnaire_sJS/";
