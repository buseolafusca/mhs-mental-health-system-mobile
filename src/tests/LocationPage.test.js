import React from 'react'
import { shallow, mount } from 'enzyme'
import LocationPage from '../layouts/LocationPage'
import { configure } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import ReactDOM from 'react-dom'
import { getLocationByPostCodeUrl } from '../variables/URLs'
import { MemoryRouter } from 'react-router'
import { Route } from 'react-router-dom'


describe('Location page', () => {
  it('should render correctly in mode', () => {
    const component = shallow(<LocationPage/>)

    expect(component).toMatchSnapshot()
  })
})


it('renders without crashing', () => {
  const div = document.createElement('div')
  ReactDOM.render(<LocationPage />, div)
  ReactDOM.unmountComponentAtNode(div)
  
})


// test('invalid path should redirect to 404', () => {
//   const wrapper = mount(
//     <MemoryRouter initialEntries={[ '/locationpage' ]}>
//       <LocationPage/>
//     </MemoryRouter>
//   );
//   // expect(wrapper.find('.SurveyResult1')).toHaveLength(1);
//   // expect(wrapper.find('.postcode')).toHaveLength(1);
//   // expect(wrapper.find('SurveyResult1')).toHaveLength(1);
//   expect(wrapper.find(Route)).toEqual(1);
// });


// it("onChange param is the same value as the input element's value property", () => {
//     const mockFn = jest.fn();
//     const input = shallow(<LocationPage handleChange={mockFn}/>);

//     input.find('input').simulate('change', {target: {value: 'matched'} });
//     expect(mockFn.mock.calls[0][0]).toBe('matched');
// });

 // LocationPage.js  |    35.71 |      100 |    33.33 |    35.71 |... 34,35,36,41,49 |

// describe('Test LocationPage willMount', () => {
//   beforeAll(() => {
    
//     nock(getLocationByPostCodeUrl)
//       .get()
//       .reply(200, {
//             'data': 1
//       });
//   });

//   it('Component fetching LandingPage from API', async (done) => {
//       const root = shallow(<LocationPage/>);
//       await waitUntil(() => root.state('questionnaireList') !== null);

      
//       // console.log("tttt")
//       // console.log(root.state)

//       // console.log(root.state())
//       // expect(root.state('questionnaireList')).toEqual([]);
//       expect(root.state('questionnaireList').length).toEqual(1);
//       done();
//   });
// });