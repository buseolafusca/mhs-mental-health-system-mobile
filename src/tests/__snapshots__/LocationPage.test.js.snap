// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Location page should render correctly in mode 1`] = `
<BrowserRouter>
  <Switch>
    <Route
      exact={true}
      path="/locationpage"
      render={[Function]}
    />
  </Switch>
</BrowserRouter>
`;
